Trying to make DysonPureCoolDevice() shorter and easier to read

jhthorsen opened the pull request:
Thanks or the Homebridge Plugin! I'm trying to make some changes to expose my Dyson as a single HeaterCooler service, instead of a AirPurifier and a Thermostat (I think). Either way, in the process I see that the DysonPureCoolDevice() is quite long, so I was wondering if you would be interested in this PR and further refactoring?

I can open an issue to further illustrate some ideas I have, but opening this PR first to see if there's any interest in the changes.

lukasroegner commented on src/productTypeInfo.js:
```diff
@@ -0,0 +1,75 @@
+  },
+};
+
+const clone = (data) => JSON.parse(JSON.stringify(data));
```
I like the idea of separating the product info from the logic code! However, the cloning approach seems a bit complicated. What do you think about using a switch:

```
switch (productType) {
  case '358':
    return {
      model: 'Dyson Pure Humidify+Cool',
      hardwareRevision: 'PH01',
      hasAdvancedAirQualitySensors: true,
      hasHumidifier: true,
      hasJetFocus: true
    };
  ...
}
```

jhthorsen replied:
Thanks for the feedback! I pushed a new change now, where I simply copy the attributes from a `knownProducts` item into a new object. The reason why I'm not a huge fan of the switch() approach is that it's a bit messy to add default values afterwards. This can be fixed however by adding another private function that returns the know attributes, and then apply the defaults afterwards. Please let me know what you prefer.

lukasroegner replied:
I like your approach, much better than cloning via JSON serialization!

lukasroegner commented:
@jhthorsen Thank you for the PR! It would be great if you want to work on refactoring the plugin. Please take a look at my comment in the `productType.js`.

jhthorsen commented:
I added another commit to this PR, which adds support for `npm run test`. I wasn't sure about the change I just did would work the same way, so therefor I added a test, instead of pushing this version to my live instance of Homebridge.

Please let me know what you think, and if you have any input on the choice of `tap`.

lukasroegner commented:
`tap` looks fine to me!

If you want to contribute further to the plugin, let me know whether I should add you as a collaborator.

jhthorsen commented:
> If you want to contribute further to the plugin, let me know whether I should add you as a collaborator.

I think I want to keep creating pull requests for now, until we know we are in sync. Feel it's a bit risky if I go all crazy without proper peer review, hehe

Thanks for accepting the PR!
